### ISSUE TITLE
Add TOC for subsections to each chapter page

### ISSUE THREAD

**Issue by niftylettuce** [select2/docs#6]
https://github.com/select2/docs/blob/develop/pages/12.programmatic-control/chapter.md

**Comment by alexweissman** [comment]
So, that page is really just a chapter heading for the subsections within it.  But, it might be helpful to have some brief text there anyway.  Any suggestions?

**Comment by niftylettuce** [comment]
It's really odd navigation.  When I go to that chapter heading it should take me to view the first subsection?  In general though, the docs are really really hard and awkward to navigate and use.

Why don't you just use https://github.com/egoist/docute?

**Comment by alexweissman** [comment]
> Why don't you just use https://github.com/egoist/docute?

Yeah, I'll just learn a new framework and completely rewrite the docs again, after having already done a rewrite to use Grav last month.

I think you'll agree, the new docs are orders of magnitude an improvement over what we had before.  If you have specific suggestions, please open a new issue or, even better, submit a PR.

One simple thing we can do, is just replicate the subsection headings as a list in the main chapter pages.

If you have complaints about the overall theme, you can take them up at the theme's repo: https://github.com/getgrav/grav-theme-learn2

**Comment by niftylettuce** [comment]
It should be a drop in replacement and would take 1 hr to learn at most...
The docs right now are really bad.  I'm just giving you honest feedback.

**Comment by alexweissman** [comment]
Great, I'll look forward to reviewing your pull request!

**Comment by niftylettuce** [comment]
Why so snarky haha, I'm def not going to PR now.  Good luck

**Comment by alexweissman** [comment]
> Why so snarky

Because you're asking me to do a major overhaul of something that isn't really that big of a problem.  Consider the number of bugs, and the ~600 open issues on Select2 itself, which need to take precedence.  I'm one dude, with a lot of other projects on my plate, and the best I can do is try to triage and review others' pull requests.

I don't know how long you've been using Select2, but up until about a month ago, the documentation was truly [disorganized, horrible, and out of date](https://web.archive.org/web/20170706155918/https://select2.github.io/examples.html).  I spent the better part of a week reorganizing and redesigning the documentation to make it usable and searchable.  

You can imagine how it must feel for someone to then come along and then say "Hey, you know that thing you spent a week of your time on, without being compensated, to help out the Select2 community?  Well it's bad and you should use X instead."

TL;DR: instead of complaining, contribute.
